core/commands/log.py

```diff
@@ -6,169 +6,74 @@
-        self.on_hash_selection(index)
-
-    def on_hash_selection(self, index):
-        options_array = [
-                "Show commit",
-                "Checkout commit",
-                "Compare commit against ...",
-                "Copy the full SHA",
-                "Diff commit",
-                "Diff commit (cached)"
-        ]
-
-        if self._log_current_file:
-            options_array.append("Show file at commit")
+        self.window.run_command("gs_show_commit_info", {"commit_hash": self._hashes[index]})
 
+    def on_commit_selection(self, index):
+        self.window.run_command("hide_panel", {"panel": "output.show_commit_info"})
         if index == -1:
             return
         if index == self._limit:
-            self._pagination += self._limit
+            self._skip += self._limit
             sublime.set_timeout_async(self.run_async, 1)
             return
+        self._selected_commit = self._hashes[index]
+        self.do_action(self._selected_commit)
 
-        self._selected_hash = self._hashes[index]
-
-        self.window.show_quick_panel(
-            options_array,
-            self.on_output_selection,
-            flags=sublime.MONOSPACE_FONT,
-            selected_index=self.quick_panel_log_idx
-        )
-
-    def on_output_selection(self, index):
-        if index == -1:
-            sublime.set_timeout_async(self.run_async, 1)
-            return
-
-        self.quick_panel_log_idx = index
-
-        if index == 0:
-            self.window.run_command("gs_show_commit", {"commit_hash": self._selected_hash})
-
-        if index == 1:
-            self.checkout_ref(self._selected_hash)
-            util.view.refresh_gitsavvy(self.view)
-
-        if index == 2:
-            self.window.run_command("gs_compare_against", {
-                "target_commit": self._selected_hash,
-                "file_path": self._filename
-            })
-
-        if index == 3:
-            sublime.set_clipboard(self._selected_hash)
-
-        if index in [4, 5]:
-            in_cached_mode = index == 5
-            self.window.run_command("gs_diff", {
-                "in_cached_mode": in_cached_mode,
-                "file_path": self._filename,
-                "current_file": bool(self._filename),
-                "base_commit": self._selected_hash,
-                "disable_stage": True
-            })
-
-        if index == 6:
-            lang = self.window.active_view().settings().get('syntax')
-            self.window.run_command(
-                "gs_show_file_at_commit",
-                {"commit_hash": self._selected_hash, "filepath": self._filename, "lang": lang})
+    def do_action(self, commit_hash):
+        self.window.run_command("gs_log_action", {
+            "commit_hash": commit_hash,
+            "file_path": self._file_path
+        })
 
 
-class GsLogCurrentFileCommand(WindowCommand, GitCommand):
+class GsLogCurrentBranchCommand(GsLogBase):
+    pass
```

Comment:
Originally, I wanted to define a base class and every command would be a subclass of it. It turns out that `GsLogCurrentBranchCommand` does not need extra definitions to work. 

One possible way is to rename `GsLogBase` as `GsLogCurrentBranchCommand`. But it seems strange to do

```python
class GsLogByBranchCommand(GsLogCurrentBranchCommand):
    ....

```
rather than

```python
class GsLogByBranchCommand(GsLogBase):
    ....

```